src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -127,6 +133,15 @@ private static void OnRequestSendRequestComplete(WinHttpRequestState state)
             state.TcsSendRequest.TrySetResult(true);
         }
 
+        private static void OnRequestDataAvailable(WinHttpRequestState state, int bytesAvailable)
+        {
+            Debug.Assert(state != null, "OnRequestDataAvailable: state is null");
```

Comment:
Consider adding an assert that bytesAvailable is > 0.  If it's possible instead that it is 0, I assume we don't want to be completing the data available task.